django/middleware/http.py

```diff
@@ -15,6 +18,9 @@ def process_response(self, request, response):
         if not response.streaming and not response.has_header('Content-Length'):
             response['Content-Length'] = str(len(response.content))
 
+        if settings.USE_ETAGS and not response.has_header('ETag'):
```

Comment:
What's the purpose of the settings check here? ETag generation will become compulsory when the deprecation is over. Are there advantages in delaying it? With the current implementation, I'm not sure it'll be possible to set `settings.USE_ETAGS=True` to opt-in here while also avoiding the other deprecation warnings guarded by `if settings.USE_ETAGS`.